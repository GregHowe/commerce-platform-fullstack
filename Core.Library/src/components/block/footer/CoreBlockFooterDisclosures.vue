<template>
	<CoreBlock
		v-if="showDisclosures"
		class="disclosures"
		:class="{ 'disclosures--mobile': isMobile }"
		:settings="accordionSettings"
	/>
</template>

<script>
import { renderData } from "@libraryHelpers/dataComponents.js";

export default {
	name: "CoreBlockFooterDisclosures",
	props: {
		site: {
			type: Object,
			required: true,
		},
		isMobile: {
			type: Boolean,
			default: false,
		},
	},
	data() {
		return {
			clonedSite: null,
			stateValues: [
				{ value: "AL", text: "Alabama" },
				{ value: "AK", text: "Alaska" },
				{ value: "AZ", text: "Arizona" },
				{ value: "AR", text: "Arkansas" },
				{ value: "CA", text: "California" },
				{ value: "CO", text: "Colorado" },
				{ value: "CT", text: "Connecticut" },
				{ value: "DE", text: "Delaware" },
				{ value: "FL", text: "Florida" },
				{ value: "GA", text: "Georgia" },
				{ value: "HI", text: "Hawaii" },
				{ value: "ID", text: "Idaho" },
				{ value: "IL", text: "Illinois" },
				{ value: "IN", text: "Indiana" },
				{ value: "IA", text: "Iowa" },
				{ value: "KS", text: "Kansas" },
				{ value: "KY", text: "Kentucky" },
				{ value: "LA", text: "Louisiana" },
				{ value: "ME", text: "Maine" },
				{ value: "MD", text: "Maryland" },
				{ value: "MA", text: "Massachusetts" },
				{ value: "MI", text: "Michigan" },
				{ value: "MN", text: "Minnesota" },
				{ value: "MS", text: "Mississippi" },
				{ value: "MO", text: "Missouri" },
				{ value: "MT", text: "Montana" },
				{ value: "NE", text: "Nebraska" },
				{ value: "NV", text: "Nevada" },
				{ value: "NH", text: "New Hampshire" },
				{ value: "NJ", text: "New Jersey" },
				{ value: "NM", text: "New Mexico" },
				{ value: "NY", text: "New York" },
				{ value: "NC", text: "North Carolina" },
				{ value: "ND", text: "North Dakota" },
				{ value: "OH", text: "Ohio" },
				{ value: "OK", text: "Oklahoma" },
				{ value: "OR", text: "Oregon" },
				{ value: "PA", text: "Pennsylvania" },
				{ value: "RI", text: "Rhode Island" },
				{ value: "SC", text: "South Carolina" },
				{ value: "SD", text: "South Dakota" },
				{ value: "TN", text: "Tennessee" },
				{ value: "TX", text: "Texas" },
				{ value: "UT", text: "Utah" },
				{ value: "VT", text: "Vermont" },
				{ value: "VA", text: "Virginia" },
				{ value: "WA", text: "Washington" },
				{ value: "WV", text: "West Virginia" },
				{ value: "WI", text: "Wisconsin" },
				{ value: "WY", text: "Wyoming" },
			],
		};
	},
	computed: {
		user() {
			return this.site.user || {};
		},
		disclosures() {
			const disclosures = this.site.footer?.disclosures || [];
			return disclosures.map((item) => renderData(item, this.clonedSite));
		},
		showDisclosures() {
			return !!this.site.footer?.showDisclosures;
		},
		accordionSettings() {
			return {
				type: "accordion",
				variants: {},
				accordionText: "Disclosure",
				hasAccordion: true,
				blocks: [
					...this.disclosures.map((disclosure) => {
						return {
							type: "rich-text",
							variants: {
								size: "disclosure",
							},
							richText: disclosure,
						};
					}),
				],
				locked: null,
			};
		},
		sellingInsuranceLicenses() {
			return this.licenses({
				licenseLobCode: "80",
				busLicenseTpCode: "I",
				busEntityCode: "001",
			});
		},
		registeredAgentLicenses() {
			return this.licenses({ licenseLobCode: "30" });
		},
		eagleAdvisorLicenses() {
			return this.licenses({ eagleData: "Y" });
		},
		statesSellingInsurance() {
			return this.states(this.sellingInsuranceLicenses);
		},
		sellingInsuranceStateText() {
			return this.stateText(this.sellingInsuranceLicenses);
		},
		statesRegisteredAgent() {
			return this.states(this.registeredAgentLicenses);
		},
		registeredAgentStateText() {
			return this.stateText(this.registeredAgentLicenses);
		},
		statesEagleFinancial() {
			return this.states(this.eagleAdvisorLicenses);
		},
		eagleFinancialStateText() {
			return this.stateText(this.eagleAdvisorLicenses);
		},
	},
	created() {
		this.addDataToSite();
	},
	methods: {
		stateText(states) {
			return states.length > 1 ? "states" : "state";
		},
		licenses(args = {}) {
			const licenses = this.user.ddcUserData?.ddcLicenseData || [];
			const filtered = licenses.filter((license) => {
				const expireDate = new Date(license.licenseExpiryDt).getTime();
				const issueDate = new Date(license.licenseIssueDt).getTime();
				const today = new Date().getTime();
				const query = Object.keys(args).every((key) => {
					return license[key] === args[key];
				});
				return query && expireDate >= today && issueDate <= today;
			});
			// only one per state (remove duplicate valid states)
			return filtered.filter(
				(obj, index, self) =>
					index ===
					self.findIndex(
						(t) => t.stateCountyCode === obj.stateCountyCode
					)
			);
		},
		states(licenses) {
			// create a string of states based on the license data
			const appendableStates = ["CA", "AR"];
			const stateArray = licenses
				.map((license) => {
					const state = license.stateCountyCode;
					// append the license number to certain states
					if (appendableStates.includes(state)) {
						return `${state} (${state} Insurance License #${license.licenseIdNumber})`;
					}
					// if the state is a number, use "FL" instead
					if (!isNaN(state)) {
						return "FL";
					}
					return state;
				})
				.join(",")
				.split(",");
			// Sort the array alphabetically
			stateArray.sort();
			// if they are licensed in more than one state, join them with "and" before the last item
			if (stateArray.length > 1) {
				const lastItem = stateArray.pop();
				return `${stateArray.join(", ")}, and ${lastItem}`;
			}

			// if they are licensed in only one state, return the full name of the state
			const singleState = stateArray[0].split(" ")[0].trim();
			const fullState = this.stateValues.find((state) => {
				return state.value === singleState;
			});
			if (fullState) {
				const license = licenses.find(
					(l) => l.stateCountyCode === fullState.value
				);
				if (
					license?.licenseIdNumber &&
					appendableStates.includes(license.stateCountyCode.trim())
				) {
					return `${fullState.text} (${fullState.text} Insurance License #${license.licenseIdNumber})`;
				}
				return fullState.text;
			}
			return "";
		},
		addDataToSite() {
			// here we can add various data points to use in the disclosure via mustache templates
			// needs to be done in created so that the computed data is available in the mustache templates
			this.clonedSite = {
				statesSellingInsurance: this.statesSellingInsurance,
				statesRegisteredAgent: this.statesRegisteredAgent,
				sellingInsuranceStateText: this.sellingInsuranceStateText,
				registeredAgentStateText: this.registeredAgentStateText,
				eagleFinancialStateText: this.eagleFinancialStateText,
				statesEagleFinancial: this.statesEagleFinancial,
				...this.site,
			};
		},
	},
};
</script>

<style lang="scss">
.disclosures {
	color: var(--core__footer-font_color);
	padding: 1.5rem 0;
	.block__body {
		font-size: var(--_core__disclosure_font-size);
	}
	&--mobile {
		padding: 0 30px;
	}
}
</style>
