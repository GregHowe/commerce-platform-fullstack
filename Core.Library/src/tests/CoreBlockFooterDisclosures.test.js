import CoreBlockFooterDisclosures from "../components/block/footer/CoreBlockFooterDisclosures.vue";
import "@testing-library/jest-dom";
import { render } from "@testing-library/vue";

const renderComponent = async (user = {}, site = {}) => {
	return render(CoreBlockFooterDisclosures, {
		props: {
			site: {
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [],
				},
				...site,
				user,
			},
		},
	});
};

describe("CoreBlockFooterDisclosures", () => {
	test("exists", async () => {
		await renderComponent();
	});

	test("disclosure shows eagle financial states and text", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					ddcLicenseData: [
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							stateCountyCode: "AR",
							licenseIdNumber: "1234",
							eagleData: "Y",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							stateCountyCode: "AL",
							licenseIdNumber: "1234",
							eagleData: "Y",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							stateCountyCode: "WA",
							licenseIdNumber: "1234",
							eagleData: "N",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesEagleFinancial }} {{ eagleFinancialStateText }} stuff even works for eagle financial</p>",
					],
				},
			}
		);
		getByText(
			"test mustache AL, and AR (AR Insurance License #1234) states stuff even works for eagle financial"
		);
		expect(container).toMatchSnapshot();
	});

	test("multiple selling licenses show comma seperated states in alphabetical order, with CA and AR including license number", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					regRepInd: "Y",
					ddcLicenseData: [
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							stateCountyCode: "AR",
							licenseIdNumber: "1234",
							eagleData: "Y",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							stateCountyCode: "AL",
							licenseIdNumber: "1234",
							eagleData: "Y",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "CA",
							licenseIdNumber: "CA1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "WA",
							licenseIdNumber: "WA1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "FL",
							licenseIdNumber: "FL1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "ND",
							licenseIdNumber: "ND1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "AR",
							licenseIdNumber: "AR1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "30",
							stateCountyCode: "ND",
							licenseIdNumber: "ND1234",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "30",
							stateCountyCode: "AR",
							licenseIdNumber: "AR1234",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesSellingInsurance }} {{ sellingInsuranceStateText }} stuff even works for selling insurance</p>",
						"<p>test mustache {{ statesRegisteredAgent }} {{ registeredAgentStateText }} stuff even works for registered agents</p>",
						"<p>test mustache {{ statesEagleFinancial }} {{ eagleFinancialStateText }} stuff even works for eagle financial</p>",
					],
				},
			}
		);
		getByText(
			"test mustache AR (AR Insurance License #AR1234), CA (CA Insurance License #CA1234), FL, ND, and WA states stuff even works for selling insurance"
		);
		getByText(
			"test mustache AR (AR Insurance License #AR1234), and ND states stuff even works for registered agents"
		);
		getByText(
			"test mustache AL, and AR (AR Insurance License #1234) states stuff even works for eagle financial"
		);
		expect(container).toMatchSnapshot();
	});

	test("only licenses within timeframe are analyzed", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					regRepInd: "Y",
					ddcLicenseData: [
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "CA",
							licenseIdNumber: "CA1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "WA",
							licenseIdNumber: "WA1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "FL",
							licenseIdNumber: "FL1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "ND",
							licenseIdNumber: "ND1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "1975-12-31",
							licenseIssueDt: "2000-12-31",
							licenseLobCode: "80",
							stateCountyCode: "AR",
							licenseIdNumber: "AR1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "30",
							stateCountyCode: "ND",
							licenseIdNumber: "ND1234",
						},
						{
							licenseExpiryDt: "1975-12-31",
							licenseIssueDt: "2000-12-31",
							licenseLobCode: "30",
							stateCountyCode: "AR",
							licenseIdNumber: "AR1234",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesSellingInsurance }} {{ sellingInsuranceStateText }} stuff even works for selling insurance</p>",
						"<p>test mustache {{ statesRegisteredAgent }} {{ registeredAgentStateText }} stuff even works for registered agents</p>",
					],
				},
			}
		);
		getByText(
			"test mustache CA (CA Insurance License #CA1234), FL, ND, and WA states stuff even works for selling insurance"
		);
		getByText(
			"test mustache North Dakota state stuff even works for registered agents"
		);
		expect(container).toMatchSnapshot();
	});

	test("shows a single licensed state of CA with the full name of the state", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					ddcLicenseData: [
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "CA",
							licenseIdNumber: "CA1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesSellingInsurance }} {{ sellingInsuranceStateText }} stuff even works for selling insurance</p>",
					],
				},
			}
		);
		getByText(
			"test mustache California (California Insurance License #CA1234) state stuff even works for selling insurance"
		);
		expect(container).toMatchSnapshot();
	});

	test("a numeric state returns as Florida for registered representative and selling insurance", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					regRepInd: "Y",
					ddcLicenseData: [
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "30",
							stateCountyCode: "123",
							licenseIdNumber: "1234",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "123",
							licenseIdNumber: "1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesSellingInsurance }} {{ sellingInsuranceStateText }} stuff even works for selling insurance</p>",
						"<p>test mustache {{ statesRegisteredAgent }} {{ registeredAgentStateText }} stuff even works for registered agents</p>",
					],
				},
			}
		);
		getByText(
			"test mustache Florida state stuff even works for selling insurance"
		);
		getByText(
			"test mustache Florida state stuff even works for registered agents"
		);
		expect(container).toMatchSnapshot();
	});

	test("state text is singular if one registered license", async () => {
		const { container, getByText } = await renderComponent(
			{
				ddcUserData: {
					ddcLicenseData: [
						{
							licenseExpiryDt: "2085-12-31",
							licenseIssueDt: "2091-12-31",
							licenseLobCode: "80",
							stateCountyCode: "AR",
							licenseIdNumber: "1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
						{
							licenseExpiryDt: "2025-12-31",
							licenseIssueDt: "2021-12-31",
							licenseLobCode: "80",
							stateCountyCode: "AL",
							licenseIdNumber: "1234",
							busLicenseTpCode: "I",
							busEntityCode: "001",
						},
					],
				},
			},
			{
				footer: {
					showFooter: true,
					showDisclosures: true,
					disclosures: [
						"<p>test mustache {{ statesSellingInsurance }} {{ sellingInsuranceStateText }} stuff even works for selling insurance</p>",
					],
				},
			}
		);
		getByText(
			"test mustache Alabama state stuff even works for selling insurance"
		);
		expect(container).toMatchSnapshot();
	});
});
