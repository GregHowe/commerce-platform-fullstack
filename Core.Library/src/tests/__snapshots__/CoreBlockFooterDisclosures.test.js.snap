// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreBlockFooterDisclosures a numeric state returns as Florida for registered representative and selling insurance 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache Florida state stuff even works for selling insurance
            </p>
          </div>
            
          <!---->
        </div>
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="1"
            nested-level="1"
          >
            <p>
              test mustache Florida state stuff even works for registered agents
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;

exports[`CoreBlockFooterDisclosures disclosure shows eagle financial states and text 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache AL, and AR (AR Insurance License #1234) states stuff even works for eagle financial
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;

exports[`CoreBlockFooterDisclosures multiple selling licenses show comma seperated states in alphabetical order, with CA and AR including license number 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache AR (AR Insurance License #AR1234), CA (CA Insurance License #CA1234), FL, ND, and WA states stuff even works for selling insurance
            </p>
          </div>
            
          <!---->
        </div>
        <div
          class="block block_rich-text block_rich-text_disclosure"
          target="_blank"
        >
          <div
            class="block__body"
            index="1"
            nested-level="1"
          >
            <p>
              test mustache AR (AR Insurance License #AR1234), and ND states stuff even works for registered agents
            </p>
          </div>
            
          <!---->
        </div>
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="2"
            nested-level="1"
          >
            <p>
              test mustache AL, and AR (AR Insurance License #1234) states stuff even works for eagle financial
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;

exports[`CoreBlockFooterDisclosures only licenses within timeframe are analyzed 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache CA (CA Insurance License #CA1234), FL, ND, and WA states stuff even works for selling insurance
            </p>
          </div>
            
          <!---->
        </div>
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="1"
            nested-level="1"
          >
            <p>
              test mustache North Dakota state stuff even works for registered agents
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;

exports[`CoreBlockFooterDisclosures shows a single licensed state of CA with the full name of the state 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache California (California Insurance License #CA1234) state stuff even works for selling insurance
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;

exports[`CoreBlockFooterDisclosures state text is singular if one registered license 1`] = `
<div>
  <div
    class="disclosures block block_accordion block_accordion_base"
    target="_blank"
  >
    <div
      class="block__body"
    >
      <!---->
       
      <!---->
       
      <div
        class="block__control block__control_accordion"
      >
        
	Disclosure
	
        <span
          icon="circle-plus"
        />
      </div>
       
      <div
        class="block__section_main"
        style="display: none;"
      >
        <div
          class="block block_rich-text block_rich-text_disclosure"
          is-first-child="true"
          is-last-child="true"
          target="_blank"
        >
          <div
            class="block__body"
            index="0"
            nested-level="1"
          >
            <p>
              test mustache Alabama state stuff even works for selling insurance
            </p>
          </div>
            
          <!---->
        </div>
      </div>
       
      <!---->
    </div>
      
    <!---->
  </div>
</div>
`;
